plugin-offline/src/main/java/com/mapbox/mapboxsdk/plugins/offline/ui/RegionSelectionFragment.java

```diff
@@ -0,0 +1,186 @@
+  MapboxMap.OnCameraIdleListener {
+
+  public static final String TAG = "OfflineRegionSelectionFragment";
+  private static final String[] LAYER_IDS = new String[] {
+    "place-city-lg-n", "place-city-lg-s", "place-city-md-n", "place-city-md-s", "place-city-sm"
+  };
+  private static final String[] SOURCE_LAYER_IDS = new String[] {
+    "place_label", "state_label", "country_label"
+  };
+
+  private RegionSelectedCallback selectedCallback;
+  private TextView regionNameTextView;
+  private MapboxMap mapboxMap;
+  private RectF boundingBox;
+  private MapView mapView;
+  private View rootView;
+  String regionName;
+
+  public static RegionSelectionFragment newInstance() {
+    return new RegionSelectionFragment();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                           @Nullable Bundle savedInstanceState) {
+    rootView = inflater.inflate(R.layout.mapbox_offline_region_selection_fragment, container, false);
+    mapView = rootView.findViewById(R.id.mapbox_offline_region_selection_map_view);
+    regionNameTextView = rootView.findViewById(R.id.mapbox_offline_region_name_text_view);
+    return rootView;
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+    bindClickListeners();
+  }
+
+  public RegionSelectedCallback getSelectedCallback() {
+    return selectedCallback;
+  }
+
+  public void setSelectedCallback(@NonNull RegionSelectedCallback selectedCallback) {
+    this.selectedCallback = selectedCallback;
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    mapboxMap.addOnCameraIdleListener(this);
+  }
+
+  @Override
+  public void onCameraIdle() {
+    if (boundingBox == null) {
+      boundingBox = getSelectionRegion();
+    }
+    Timber.v("Camera moved");
+    regionName = getOfflineRegionName();
+    regionNameTextView.setText(regionName);
+  }
+
+  @Override
+  public void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onSaveInstanceState(@NonNull Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public void onDestroyView() {
+    super.onDestroyView();
+    mapView.onDestroy();
+  }
+
+  private RectF getSelectionRegion() {
+    View selectionBoxView = rootView.findViewById(R.id.mapbox_offline_scrim_view);
```

Comment:
since getSelectionRegion is called often, we could keep this view in memory

Reply:
It should only be called once if boundingbox is null and when we create the region definition. For some reason when placed in `onViewCreated` the view is still measured at 0.